### ISSUE TITLE
Hide the Cell tab when the structure is not periodic

### ISSUE THREAD

**Issue by unkcpz** [aiidalab/aiidalab-widgets-base#351]
The `StructureDataviewer` will show the cell tab if the cell tab is initialized when creating the viewer. 
It makes no sense to show the cell information for the non-periodic structure. 
So when the structure detected is not 0-D structure the `cell` tab should be completely hidden. 

But for the 1-D and 2-D structures, there are options here, either we can hide the cell tab completely or we can show partial information with the cell.

**Comment by AndresOrtegaGuerrero** [comment]
#488 address this ? , if you test now the viewer do not hide the cell parameters for 1D and 2D periodicity

**Comment by danielhollas** [comment]
@AndresOrtegaGuerrero are you sure? I think this issue is about something else than what was addressed in #488. Here the idea was that the Cell tab should be completely hidden when the structure is not periodic. I don't think that's the case now?

**Comment by AndresOrtegaGuerrero** [comment]
@danielhollas if the behaviour desired is like that , then no, it doesnt hide the cell tab when having a 0D material, but regarding 1D and 2D materials it doesnt disappear the cell parameters

**Comment by unkcpz** [comment]
Emm... @danielhollas is right, I didn't read carefully what I wrote before, sorry. If the structure is a molecule, the tab will still show. But the behavior is more confusing now. I upload a water molecule and there are values for the cell.  Not sure where it comes from.
![Peek 2023-09-26 17-56](https://github.com/aiidalab/aiidalab-widgets-base/assets/8831685/ef629bfd-8c52-4ed2-bbe3-8e5202aa037d)

**Comment by danielhollas** [comment]
I think that's because we add a dummy cell to gas phase molecule. In the old AiiDA, there needed to be default cell, otherwise StructureData would not work. But that has been fixed so we should get rid of this login in AWB.

https://github.com/aiidalab/aiidalab-widgets-base/blob/master/aiidalab_widgets_base/structures.py#L379

**Comment by danielhollas** [comment]
The fix in aiida-core was done here https://github.com/aiidateam/aiida-core/pull/5341
and released in version 2.0 https://github.com/aiidateam/aiida-core/blob/main/CHANGELOG.md#v200---2022-04-27

**Comment by danielhollas** [comment]
@unkcpz can you try if it works correctly if you generate a molecule from SMILES?

**Comment by unkcpz** [comment]
I'll try it later, I only have QE app at hand which doesn't have SMILES tab open :joy:

**Comment by AndresOrtegaGuerrero** [comment]
I think a future PR should be to add a widget to a cell editor, in such a way that first it can change and set the periodicity of the system. Given this widget it should control if the cell appears or disappear (It could also just set the layout of the cell tab to say the system has periodicity none, but display the symmetry information, relevant for the molecule) based on the periodicity, and then the user can set the cell parameters.

**Comment by unkcpz** [comment]
> @unkcpz can you try if it works correctly if you generate a molecule from SMILES?

![image](https://github.com/aiidalab/aiidalab-widgets-base/assets/8831685/652a23ce-d31b-47e1-b869-12236885dd79)

I think it is all fine. But for the SMILE widget itself, I first encounter `rdkit` is not installed and then after I install the rdkit from conda, I encountered `sklearn` not installed. 
I think we need to make those extra dependencies for smile.

**Comment by unkcpz** [comment]
>  But that has been fixed so we should get rid of this login in AWB.

@danielhollas I think we can safely remove this function?

**Comment by danielhollas** [comment]
Yes, I think we should remove that function. btw: SMILES-generated molecules also should not have any cell, not sure where that comes from.

> But for the SMILE widget itself, I first encounter rdkit is not installed and then after I install the rdkit from conda, I encountered sklearn not installed. I think we need to make those extra dependencies for smile.

That's already done in `setup.cfg` as extra dependencies (RDkit and sklearn are big so we do not want them to be installed by default. You can install them as `pip install .[smiles]`

**Comment by unkcpz** [comment]
I thought this was not hard to solve, but after some attempts, this requires more tidy up on the `StrucutreDataViewer` and therefore goes with other on going PRs related to structure viewer. 
I move this issue to milestone `v2.2.0`.
